SimulationRuntime/c/util/modelica_string.c

```diff
@@ -199,6 +199,15 @@ modelica_string modelica_real_to_modelica_string_format(modelica_real r,modelica
   return res;
 }
 
+/* Convert a modelica_string to a modelica_string, used in String(s) */
+
+modelica_string modelica_string_to_modelica_string(modelica_string s)
+{
+  void *res = alloc_modelica_string(MMC_STRLEN(s));
+  strcpy(MMC_STRINGDATA(res), MMC_STRINGDATA(s));
+  return res;
+}
```

Comment:
Why not just return s? It's garbage collected, so this should not be a problem (or changing String(s) to just be a no-op?)